lookout/style/format/feature_extractor.py

```diff
@@ -448,210 +445,223 @@ def _create_neighbours(self, vnodes: Sequence[VirtualNode], vnodes_y: Sequence[V
+                    start, stop, current_class_seq = None, None, []
+                file.add(annotations[AtomicTokenAnnotation].to_token_annotation())
+                if ClassAnnotation in annotations:
+                    file.add(annotations[ClassAnnotation].to_target_annotation())
             else:
                 if not current_class_seq:
-                    start = vnode.start
-                end = vnode.end
-                value += vnode.value
-                if not vnode.is_accumulated_indentation:
-                    current_class_seq.extend(vnode.y)
+                    start = annotations.start
+                stop = annotations.stop
+                if not acc_indent:
+                    current_class_seq.extend(annotations[ClassAnnotation].cls)
         if current_class_seq:
-            assert value
-            yield from _class_seq_to_vnodes(value, start, end, current_class_seq, path)
+            file.add(*_class_seq_to_annotations(start, stop, current_class_seq))
 
-    def _add_noops(self, vnodes: Sequence[VirtualNode], path: str, index_labels: bool = False,
-                   ) -> List[VirtualNode]:
+    def _add_noops(self, file: AnnotationManager) -> None:
         """
-        Add CLS_NOOP nodes in between tokens without labeled nodes to allow for insertions.
+        Add `TokenAnnotation` with zero length in between `TokenAnnotation` without labeled nodes.
 
-        :param vnodes: The sequence of `VirtualNode`-s to augment with noop nodes.
-        :param path: path to file.
-        :param index_labels: Whether to index labels to define output classes or not.
-        :return: The augmented `VirtualNode`-s sequence.
+        Such zero length annotations means that some formatting sequence can be inserted to the
+        annotation position.
+
+        :param file: Source code annotated with `TokenAnnotation` and `LabelAnnotation`.
         """
-        augmented_vnodes = []
-        noop_label = (CLASS_INDEX[CLS_NOOP],)
-        if not len(vnodes):
-            return augmented_vnodes
-        if vnodes[0].y is None:
-            augmented_vnodes.append(VirtualNode(value="", start=Position(0, 1, 1),
-                                                end=Position(0, 1, 1), y=noop_label, path=path))
-        for vnode, next_vnode in zip(vnodes, islice(vnodes, 1, None)):
-            augmented_vnodes.append(vnode)
-            if vnode.y is None and next_vnode.y is None:
-                augmented_vnodes.append(VirtualNode(value="", start=vnode.end, end=vnode.end,
-                                                    y=noop_label, path=path))
-        augmented_vnodes.append(next_vnode)
-        if augmented_vnodes[-1].y is None:
-            augmented_vnodes.append(VirtualNode(value="", start=vnodes[-1].end, end=vnodes[-1].end,
-                                                y=noop_label, path=path))
-        return augmented_vnodes
+        noop_target = (CLASS_INDEX[CLS_NOOP],)
```

Comment:
```suggestion
        noop_target = CLASS_INDEX[CLS_NOOP],
```

Reply:
well, I do not like it because a coma in the end is hard to see. Brackets make it clear that it is a tuple

Reply:
IF we had a style-analyzer working with Python, you would have to change :smiling_imp: